Python/ceval.c

```diff
@@ -3650,6 +3649,43 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, _PyInterpreterFrame *frame, int
             NOTRACE_DISPATCH();
         }
 
+        TARGET(LOAD_ATTR_PROPERTY) {
+            assert(cframe.use_tracing == 0);
+            _PyLoadMethodCache *cache = (_PyLoadMethodCache *)next_instr;
+
+            PyObject *owner = TOP();
+            PyTypeObject *cls = Py_TYPE(owner);
+            uint32_t type_version = read_u32(cache->type_version);
+            DEOPT_IF(cls->tp_version_tag != type_version, LOAD_ATTR);
+            assert(type_version != 0);
+            PyObject *fget = read_obj(cache->descr);
+            PyFunctionObject *f = (PyFunctionObject *)fget;
+            DEOPT_IF(f->func_version != cache->keys_version[0], LOAD_ATTR);
```

Comment:
Why `cache->keys_version[0]` rather than `read_u32(cache->keys_version)`?

Reply:
`func_version` is stored as a `uint_16t` in cache. I don't need the entire `keys_version`. Though I can store the whole `uint32_t` form of `func_version` if you'd prefer.

Reply:
I'm curious why you chose to use a 16 bit value, rather than the full 32 bits available?

Reply:
I was thinking it'll make things easier when we want to shrink the cache yet again.

Reply:
The problem with shrinking the caches is that we probably want about 20 bits of information. 16 is probably too few, 32 too many. So, let's keep things consistent for now, as we might want to do something more complex than simply reduce the number of bits from 32 to 16.